packages/api/src/utils/client/httpClient.ts

```diff
@@ -49,62 +68,167 @@ export type HttpClientModules = {
 };
 
 export class HttpClient implements IHttpClient {
-  readonly baseUrl: string;
-  private readonly timeoutMs: number;
-  private readonly bearerToken?: string;
+  private readonly globalTimeoutMs: number;
   private readonly getAbortSignal?: () => AbortSignal | undefined;
   private readonly fetch: typeof fetch;
   private readonly metrics: null | Metrics;
   private readonly logger: null | ILogger;
 
+  private readonly urlOpts: URLOpts[] = [];
+  private readonly urlScore: number[];
+
+  get baseUrl(): string {
+    return this.urlOpts[0].baseUrl;
+  }
+
   /**
    * timeoutMs = config.params.SECONDS_PER_SLOT * 1000
    */
   constructor(opts: HttpClientOptions, {logger, metrics}: HttpClientModules = {}) {
-    this.baseUrl = opts.baseUrl;
-    // A higher default timeout, validator will sets its own shorter timeoutMs
-    this.timeoutMs = opts.timeoutMs ?? 60_000;
-    this.bearerToken = opts.bearerToken;
+    if (opts.baseUrl) {
+      this.urlOpts.push({
+        baseUrl: opts.baseUrl,
+        bearerToken: opts.bearerToken,
+        timeoutMs: opts.timeoutMs,
+      });
+    }
+
+    if (opts.urls) {
```

Comment:
should we do some filters to avoid duplicate in `this.urlOpts`?

Reply:
Yes good point, added